api/client/sg.cc

```diff
@@ -481,7 +473,61 @@ static int sub_sg_rule_del(int argc, char **argv,
 
     proto::Messages delete_res;
     return request(delete_req, &delete_res, options, false) ||
-        check_request_result(delete_res);
+            check_request_result(delete_res);
+}
+static int sub_sg_param_rule_del(int argc, char **argv,
+                                 const GlobalOptions &options) {
+    RuleAddOptions opts;
+    if (opts.parse(argc, argv)) {
+        sub_sg_rule_del_help();
+        return 1;
+    }
+
+    string req =
+        "messages {"
+        "  revision: " PROTO_REV
+        "  message_0 {"
+        "    request {"
+        "      sg_rule_del {"
+        "        sg_id: \"" + opts.sg + "\""
+        "        rule {"
+        "          direction: " + opts.direction +
+        "          protocol: " + to_string(opts.proto);
+    if (opts.has_port_start) {
+        req += "   port_start: " + to_string(opts.port_start) +
+            "   port_end: " + to_string(opts.port_end);
+    }
+    if (opts.cidr.length())
+        req += "   cidr { " + opts.cidr + " }";
+    else if (opts.sg_members.length())
+            req += "   security_group: \"" + opts.sg_members + "\"";
+    req +=
+            "        }"
+            "      }"
+            "    }"
+            "  }"
+            "}";
+
+    proto::Messages res;
+    return request(req, &res, options, false) || check_request_result(res);
+}
+static int sub_sg_rule_del(int argc, char **argv,
+                            const GlobalOptions &options) {
+    if (argc >= 5 && strcmp(argv[4], "help") == 0) {
+        sub_sg_rule_add_help();
+        return 0;
+    }
+
+    if (argc <= 5) {
+        sub_sg_rule_del_help();
+        return 1;
+    }
+    string sg = argv[4];
+    string param = argv[5];
+    if (check_hash(param) == false) {
+        return sub_sg_hashed_rule_del(sg, param, options);
+    }
+    return sub_sg_param_rule_del(argc, argv, options);
```

Comment:
May `sub_sg_hashed_rule_del` and `sub_sg_param_rule_del` have the same signature ?